### DISCUSSION TITLE
How to update the containers?

### DISCUSSION THREAD

**Discussion by sticky99** [wazuh/wazuh-docker#735] [Q&A]
I installed the Manager, Dashboard, and Indexer as a single node (v4.3.5) a few months back, and installed the same version agent on a few devices on my home network. Now the agents are updating (semi-automatically) but the docker containers are not. I generally use Watchtower for container updates, but from debugs I can see it just checks if there's a more recent 4.3.5 version available, rather than looking for a more recent version number. 

So is there a recommended update procedure, or is just to do a "docker rm" and "docker pull" and everything should work as before?

**Comment by vcerenu** [comment]
Hello @sticky99 

If you have installed it from our original YAMLs and you are on the same base version of the repository (4.3) you can update the repository, change to the most current tag, and start the stack again, since the volumes that will be used are the same.

```BASH
cd single-node
docker-compose down
cd ..
git pull
git checkout v4.3.9
cd single-node
docker-compose up -d
```

If you have made any changes to the directories or have your own YAML to deploy, modifying the version of each image in the docker-compose.yml is enough.

**Reply by cb3inco** [reply]
@vcerenu 

I'm following these directions, running the following command, but the v4.3.10 doesn't seem to be working

sudo git checkout v4.3.10
error: pathspec 'v4.3.10' did not match any file(s) known to git

I did do the sudo git pull it says it's all up to date. If I look at the VERSION file, it says:

WAZUH-DOCKER_VERSION="4.3.9"
REVISION="40322"

Thanks.

**Reply by cb3inco** [reply]
So, I think the problem is with the way docker is to be installed.  Per the documentation it says to do this:

git clone https://github.com/wazuh/wazuh-docker.git -b v4.3.9 --depth=1

Which then writes into the ./git/config file the following:

[remote "origin"]
        url = https://github.com/wazuh/wazuh-docker.git
        fetch = +refs/tags/v4.3.9:refs/tags/v4.3.9

The problem then is when you do a git pull, git checkout v4.3.10 it errors out like I've show above.

I ended up modifying the ./git/config file to reflect the following:

[remote "origin"]
        url = https://github.com/wazuh/wazuh-docker.git
        fetch = +refs/tags/v4.3.10:refs/tags/v4.3.10

Once I did this, I was able to do the pull and checkout with no problem. so it would seem these instructions are incomplete. I'm sure there is another/better way of doing this, but I don't know. I got it to update, so it's fine this time.
